ui/src/pages/AppRouterView.vue

```diff
@@ -22,42 +22,35 @@
 </template>
 
 <script lang="ts" setup>
-  import { PGlobalSidebar, PIcon, media } from '@prefecthq/prefect-design'
-  import { createApi, workspaceApiKey, createCan, canKey as designCanKey, workspacePermissions, createWorkspaceRoutes, workspaceRoutesKey } from '@prefecthq/prefect-ui-library'
-  import { computed, provide, ref, watchEffect } from 'vue'
+  import { PGlobalSidebar, PIcon, media, showToast } from '@prefecthq/prefect-design'
+  import { workspaceApiKey, canKey as designCanKey, createWorkspaceRoutes, workspaceRoutesKey } from '@prefecthq/prefect-ui-library'
+  import { computed, provide, watchEffect } from 'vue'
   import { RouterView } from 'vue-router'
   import ContextSidebar from '@/components/ContextSidebar.vue'
-  import { UiSettings } from '@/services/uiSettings'
+  import { useApiConfig } from '@/compositions/useApiConfig'
+  import { useCreateCan } from '@/compositions/useCreateCan'
+  import { useMobileMenuOpen } from '@/compositions/useMobileMenuOpen'
+  import { createPrefectApi, prefectApiKey } from '@/utilities/api'
   import { canKey } from '@/utilities/permissions'
 
-  const baseUrl = await UiSettings.get('apiUrl')
-  const api = createApi({
-    baseUrl,
-  })
-  const flags = await UiSettings.get('flags')
-
-  const can = createCan([
-    ...workspacePermissions,
-    ...flags,
-  ])
-
+  const { can } = useCreateCan()
+  const { config } = await useApiConfig()
+  const api = createPrefectApi(config)
   const routes = createWorkspaceRoutes()
 
   provide(canKey, can)
   provide(designCanKey, can)
+  provide(prefectApiKey, api)
   provide(workspaceApiKey, api)
   provide(workspaceRoutesKey, routes)
 
-  const mobileMenuOpen = ref(false)
-  const showMenu = computed(() => media.lg || mobileMenuOpen.value)
-
-  function toggle(): void {
-    mobileMenuOpen.value = !mobileMenuOpen.value
+  const healthy = await api.health.checkHealth()
+  if (!healthy) {
+    showToast(`Can't connect to Server API at ${config.baseUrl}. Check that it's accessible from your machine.`, 'error', { timeout: false })
```

Comment:
This needs to be reworked a little. It was intentionally in a utility before so that it wouldn't block the app rendering. As is nothing in the app will start rendering until this health check is done. We want it to be fire and forget. This might apply to the nebula version as well.

Reply:
hows this?
```ts
  api.health.isHealthy().then(healthy => {
    if (!healthy) {
      showToast(`Can't connect to Server API at ${config.baseUrl}. Check that it's accessible from your machine.`, 'error', { timeout: false })
    }
  })
```

Reply:
Yep that works. Cannot believe you're using promise.then 😱 lol

Reply:
Lol. They have their place